Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+
+    $(ul).mouseover (function (event)
+    {
+      var eventTarget = DropDownMenu.GetTarget (event, "LI");
+      $("li", ul).removeClass(DropDownMenu._itemSelectedClassName);
+      if (eventTarget.firstChild != null && eventTarget.firstChild.nodeName.toLowerCase() === 'a')
+      {
+        $(eventTarget).addClass (DropDownMenu._itemSelectedClassName);
+        (eventTarget.firstChild as HTMLElement).focus();
+      }
+    }).keydown (function (event)
+    {
+      // TODO: missing nullcheck?
+      DropDownMenu.Options_OnKeyDown (event, DropDownMenu._currentMenu!);
+    });
+
+    DropDownMenu._itemClickHandler = function(this: HTMLAnchorElement)
+    {
+      if (this.href == null || this.href === '')
+        return false;
+
+      DropDownMenu._itemClicked = true;
+      DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+      try
+      {
+        eval(this.getAttribute('javascript')!);
+      }
+      catch (e)
+      {
+      }
+      setTimeout (function () { DropDownMenu._itemClicked = false; }, 10);
+      return false;
+    };
+
+    for (let i = 0; i < itemInfos.length; i++)
+    {
+      var item = DropDownMenu.CreateItem(itemInfos[i], selectionCount);
+      if (item != null)
+        ul.appendChild(item);
+    }
+
+    var titleDivFunc = DropDownMenu.CreateTitleDivGetter (context);
+    DropDownMenu.ApplyPosition ($(div), evt, titleDivFunc());
+    $(div).iFrameShim({ top: '0px', left: '0px', width: '100%', height: '100%' });
+
+    if (DropDownMenu._repositionTimer) 
+      clearTimeout(DropDownMenu._repositionTimer);
+    var repositionHandler = function ()
+    {
+      if (DropDownMenu._repositionTimer)
+        clearTimeout (DropDownMenu._repositionTimer);
+
+      if (DropDownMenu._currentPopup && DropDownMenu._currentPopup == div && $(div).is (':visible'))
+      {
+        DropDownMenu.ApplyPosition ($(div), null, titleDivFunc());
+        DropDownMenu._repositionTimer = setTimeout (repositionHandler, DropDownMenu._repositionInterval);
+      }
+    };
+
+    // Only reposition if opened via titleDiv
+    if (evt == null)
+      DropDownMenu._repositionTimer = setTimeout(repositionHandler, DropDownMenu._repositionInterval);
+
+    if ($menuButton.data(DropDownMenu._button_timestampDataKey))
+    {
+      $menuButton.removeData(DropDownMenu._button_timestampDataKey);
+    }
+    else
+    {
+      DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+    }
+  }
+
+  private static CreateTitleDivGetter ($context: JQuery): DropDownMenu_TitleDivGetter
+  {
+    const context = $context[0];
+    const contextID = context.id;
+    if (contextID == null)
+    {
+      return function ()
+      {
+        // TODO: nullcheck?
+        return $(context.firstElementChild!);
+      };
+    }
+    else
+    {
+      return function ()
+      {
+        // TODO: shouldn't this be rewritten anyway? getElementById should only be called once here.
+        // TODO: nullability
```

Comment:
The first part does a null check and the second one returns the never-null firstchild. At least, that's the intention of the code.

Reply:
Right, if the `null` check "fails", `null` is returned.

Reply:
Result of a discussion with MK: we can remove the first `document.getElementById` but we should create an issue here to investigate if returning `null` is something that should be supported